specs/mgt-file-picker.md

```diff
@@ -1,57 +1,75 @@
+| **v1** | | |
 | `files` | `files` | An array of files to get or set the list of files rendered by the component. Use this to access the files loaded by the component. Set this value to load your own files. |
 | `insight-type` | `insightType` | Set to show the user’s trending, used, or shared files. |
+| `show-max` | `showMax` | A number value to indicate the maximum number of files to show. |
+| **v2** | | |
+| `file-list-query` | `fileListQuery` | The full query or path to the folder in a drive or site to retrieve the list of files from |
+| `file-queries` | `fileQueries` | An array of file queries to be rendered by the component |
 | `drive-id` | `driveId` | Id of the drive the folder belongs to. Must also provide either `item-id` or `item-path`. |
 | `group-id ` | `groupId` | Id of the group the folder belongs to. Must also provide either `item-id` or `item-path`. |
 | `site-id` | `siteId` | Id of the site the folder belongs to. Must also provide either `{item-id}` or `{item-path}`. Provide `{list-id}` if you’re referencing a file from a list. |
 | `item-id` | `itemId` | Id of the folder. Default query is `/me/drive/items`. Provide `{drive-id}`, `{group-id}`, `{site-id}`, or `{user-id}` to query a specific location. |
 | `item-path` | `itemPath` | Item path of the folder. Default query is `/me/drive/root`. Provide `{drive-id}`, `{group-id}`, `{site-id}`, or `{user-id}` to query a specific location. |
-| `show-max` | `showMax` | A number value to indicate the maximum number of files to show. |
 | `selected-files` | `selectedFiles`| An array of selected file objects. |
 | `multiselect` | `multiselect` | Enable/Disable selection of more than 1 file. Default is `false`.|
 
 ## Events
 
 | Event | Description |
 | ----- | ----------- |
-|selectionChanged | Fired when the user selects or deselects a file |
+| filesSelected | Fired when the user selects a file |
 
 ## APIs and Permission
 
-Same as [mgt-file-list](./mgt-file-list.md).
+| Query | Use if | Permission Scopes |
+| ----- | ------ | ----------------- |
+| **v1** | | |
+| `GET /me/insights/trending` | `insight-type` is trending | Sites.Read.All |
+| `GET /me/insights/used` | `insight-type` is `used` | " |
+| `GET /me/insights/shared` | `insight-type` is shared | " |
+| **v2** | | |
+| `GET /me/drive/root/children`	| Default (no identifiers or query provided) | Files.Read, Files.Read.All, Sites.Read.All |
+| `GET /drives/{drive-id}/items/{item-id}/children` | `{drive-id}` AND `{item-id}` | " |
+| `GET /groups/{group-id}/drive/items/{item-id}/children` | `{group-id}` AND `{item-id}` | " |
+| `GET /me/drive/items/{item-id}/children` | ONLY `{item-id}` | " | 
+| `GET /sites/{site-id}/drive/items/{item-id}/children` | `{site-id}` AND `{item-id}` | " |
+| `GET /users/{user-id}/drive/items/{item-id}/children` | `{user-id}` AND `{item-id}` | " |
+| `GET /drives/{drive-id}/root:/{path-relative-to-root}:/children` | `{drive-id}` AND `{path-relative-to-root}` | " |
 
 ## Templates
```

Comment:
It might be nice to have a `header` template that includes the top part of the picker (everything above the file list). This would give developers an easy way to style the picker (`button`, `header`, and `file`) without overriding the `default` template, which incurs extra effort on the developer.

Reply:
We should just make sure that we include a public method to spawn the big picker, in case the dev overrides the header (which includes the button to show the big picker)